File: app/src/main/java/org/sco/movieratings/MoviePreviewAdapter.java
package org.sco.movieratings;

import java.util.ArrayList;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import org.sco.movieratings.data.models.Preview;

public class MoviePreviewAdapter extends RecyclerView.Adapter<MoviePreviewAdapter.ViewHolder> {
    private static final String LOG_TAG = MoviePreviewAdapter.class.getSimpleName();

    private ArrayList<Preview> mDataset;

    public class ViewHolder extends RecyclerView.ViewHolder {
        private ImageView playButton;
        private TextView trailer;

        public ViewHolder(View v) {
            super(v);
            playButton = (ImageView) v.findViewById(R.id.playButton);
            trailer = (TextView) v.findViewById(R.id.trailerName);
        }
    }

    public MoviePreviewAdapter(ArrayList<Preview> myDataset) {
        this.mDataset = myDataset;
        notifyDataSetChanged();
    }

    public void add(Preview item) {
        mDataset.add(item);
        notifyItemInserted(mDataset.size() - 1);
    }

    public void clear() {
        mDataset.clear();
        notifyDataSetChanged();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.preview_list_item, parent, false);
        ViewHolder vh = new ViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        final Preview previewResult = mDataset.get(position);

        holder.trailer.setText(previewResult.getName());

        holder.playButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(view.getContext(),"Play Preview", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }

}

File: app/src/main/java/org/sco/movieratings/moviedetails/MoviePreviewAdapter.kt
package org.sco.movieratings.moviedetails

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import org.sco.movieratings.R
import org.sco.movieratings.api.response.MoviePreview

class MoviePreviewAdapter :
    RecyclerView.Adapter<MoviePreviewAdapter.MoviePreviewViewHolder>() {

    var moviePreviews: List<MoviePreview> = emptyList()
    var previewClickListener: (MoviePreview) -> Unit = {}

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MoviePreviewViewHolder {
        return MoviePreviewViewHolder(LayoutInflater.from(parent.context)
            .inflate(R.layout.preview_list_item, parent, false))
    }

    override fun getItemCount(): Int {
        return moviePreviews.size
    }

    override fun onBindViewHolder(holder: MoviePreviewViewHolder, position: Int) {
        if (moviePreviews.isNotEmpty()) {
            val moviePreview: MoviePreview = moviePreviews[position]
            holder.trailer.text =
                holder.itemView.resources.getString(R.string.preview_title, moviePreview.site, moviePreview.name)
            holder.playButton.setOnClickListener {
                previewClickListener(moviePreview)
            }
        }
    }

    class MoviePreviewViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var playButton: ImageView = view.findViewById(R.id.playButton)
        var trailer: TextView = view.findViewById(R.id.trailerName)
    }
}